webdriver/tests/bidi/browsing_context/set_viewport/set_viewport.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+
+from tests.support.image import png_dimensions
+
+from . import layout_viewport_dimensions
+
+@pytest.mark.asyncio
+async def test_capture(bidi_session, top_context, inline):
+    original_viewport = await layout_viewport_dimensions(bidi_session, top_context)
+
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url="about:blank", wait="complete"
+    )
+
+    await bidi_session.browsing_context.set_viewport(
+        context=top_context["context"],
+        viewport={
+          "width": 250,
+          "height": 300,
+      })
```

Comment:
So https://pptr.dev/api/puppeteer.page.setviewport suggests to set the viewport dimensions before the navigation. So it means this restriction doesn't actually exist, right and the viewport will immediately be updated and accordingly the layout gets flushed?

Reply:
I think that refers to websites that read viewport dimensions once upon load and then render things depending on the initial viewport. So although the browser would update everything as expected, the site might not catch up with this without a reload.